compiler/rustc_trait_selection/src/solve/trait_goals/structural_traits.rs

```diff
@@ -35,6 +35,8 @@ pub(super) fn instantiate_constituent_tys_for_auto_trait<'tcx>(
         | ty::Alias(ty::Projection, ..)
         | ty::Placeholder(..) => Err(NoSolution),
 
+        ty::Alias(ty::Weak, alias) => Ok(vec![tcx.type_of(alias.def_id).subst(tcx, alias.substs)]),
```

Comment:
this should be `Err(NoSolution)` similar to `ty::Alias(ty::Projection, ..)`. We should get the structural components by looking at the normalized alias